dissect/target/filesystems/jffs.py

```diff
@@ -0,0 +1,124 @@
+            c_jffs2.c_jffs2.JFFS2_MAGIC_BITMASK,
+            c_jffs2.c_jffs2.JFFS2_OLD_MAGIC_BITMASK,
+        )
+
+    def get(self, path: str) -> FilesystemEntry:
+        return JFFSFilesystemEntry(self, path, self._get_node(path))
+
+    def _get_node(self, path: str, node: Optional[jffs2.INode] = None):
+        try:
+            return self.jffs2.get(path, node)
+        except jffs2.FileNotFoundError as e:
+            raise FileNotFoundError(path, cause=e)
+        except jffs2.NotADirectoryError as e:
+            raise NotADirectoryError(path, cause=e)
+        except jffs2.NotASymlinkError as e:
+            raise NotASymlinkError(path, cause=e)
+        except jffs2.Error as e:
+            raise FileNotFoundError(path, cause=e)
+
+
+class JFFSFilesystemEntry(FilesystemEntry):
+    def _resolve(self) -> FilesystemEntry:
+        if self.is_symlink():
+            return self.readlink_ext()
+        return self
+
+    def get(self, path: str) -> FilesystemEntry:
+        entry_path = fsutil.join(self.path, path, alt_separator=self.fs.alt_separator)
+        entry = self.fs._get_node(path, self.entry)
+        return JFFSFilesystemEntry(self.fs, entry_path, entry)
+
+    def open(self) -> BinaryIO:
+        if self.is_dir():
+            raise IsADirectoryError(self.path)
+        return self._resolve().entry.open()
+
+    def iterdir(self) -> Iterator[str]:
+        if not self.is_dir():
+            raise NotADirectoryError(self.path)
+
+        if self.is_symlink():
+            for name, entry in self.readlink_ext().iterdir():
+                yield name, entry
+        else:
+            for name, entry in self.entry.iterdir():
+                yield name, entry
+
+    def scandir(self) -> Iterator[FilesystemEntry]:
+        for name, entry in self.iterdir():
+            entry_path = fsutil.join(self.path, name, alt_separator=self.fs.alt_separator)
+            yield JFFSFilesystemEntry(self.fs, entry_path, entry)
+
+    def is_dir(self, follow_symlinks: bool = False) -> bool:
+        try:
+            return self._resolve().entry.is_dir()
+        except FilesystemError:
+            return False
+
+    def is_file(self, follow_symlinks: bool = False) -> bool:
+        try:
+            return self._resolve().entry.is_file()
+        except FilesystemError:
+            return False
+
+    def is_symlink(self) -> bool:
+        return self.entry.is_symlink()
+
+    def readlink(self) -> str:
+        if not self.is_symlink():
+            raise NotASymlinkError()
+
+        return self.entry.link
+
+    def stat(self, follow_symlinks: bool = False) -> fsutil.stat_result:
+        return self._resolve().lstat()
+
+    def lstat(self) -> fsutil.stat_result:
+        node = self.entry.inode
+
+        # mode, ino, dev, nlink, uid, gid, size, atime, mtime, ctime
+        st_info = fsutil.stat_result(
+            [
+                self.entry.mode,
+                self.entry.inum,
+                0,
+                0,
```

Comment:
```suggestion
                1,
```
You could calculate a proper `nlink` number in dissect.jffs by keeping track of `dirent.ino` in a new dict and resetting the number to `0` when there's an unlink. Then expose that value in a new `nlink` property on the `INode` class.

I think it's also fine to "lie" for now and just say it's 1. If you don't decide to implement this properly, can you add a comment that it could be done in the future? Then again it should be a small addition to dissect.jffs which is still in review anyway.

Reply:
Changed to `1` for now until we implement the proper `nlink` counting in `dissect.jffs`.